### PR TITLE
gnuplot4 (last stable: 4.6.6)

### PR DESCRIPTION
**Pull request by ppannuto**
This adds the last stable version of gnuplot4. Gnuplot has many breaking
changes in syntax and behavoir between 4 and 5, which means for folks
using it, having a gnuplot4 and gnuplot5 binary around for a while is
going to be necessary for the transition period for a while.

I'm a recent port convert and have been using homebrew for all of ~36
hours. I pulled this formula from the the main repo right before it
switched over to gnuplot 5.0. I think I modified it correctly for the
versions repo; let me know if I need to fix anything.

### CODE REVIEW

File: gnuplot4.rb [hunk 1]
```diff
@@ -0,0 +1,122 @@
+class LuaRequirement < Requirement
+  fatal true
+  default_formula 'lua'
```
Comment (DomT4) [inline gnuplot4.rb]: Double quotes everywhere.

File: gnuplot4.rb [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  satisfy { which 'lua' }
+end
+
+class Gnuplot < Formula
+  homepage 'http://www.gnuplot.info'
+  url 'https://downloads.sourceforge.net/project/gnuplot/gnuplot/4.6.6/gnuplot-4.6.6.tar.gz'
+  sha256 '1f19596fd09045f22225afbfec11fa91b9ad1d95b9f48406362f517d4f130274'
+
+  # Currently, version bottles aren't hosted; that feels a little short-sighted since latest
```
Comment (DomT4) [inline gnuplot4.rb]: Kill this block. We do provide bottles for Versions' formulae now - It's a recent thing, which is why only a limited number of formulae have bottles presently.

File: gnuplot4.rb [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  # stable major version builds are so useful. Leaving the refs to the old bottles here.
+
+  #bottle do
+  #  revision 1
+  #  sha1 "c6a2e3f30495c1bd790ea5091f40b7644d695112" => :yosemite
+  #  sha1 "03d507d87eedd8c4bf3e460931081a10403f379d" => :mavericks
+  #  sha1 "24618fd48a6d5fa2f69843da8ac2aaa8d631ff48" => :mountain_lion
+  #end
+
+  option 'pdf',    'Build the PDF terminal using pdflib-lite'
```
Comment (DomT4) [inline gnuplot4.rb]: Make all of these ` with-pdf `, ` with-wx ` and so on please.

File: gnuplot4.rb [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  option 'wx',     'Build the wxWidgets terminal using pango'
```
Comment (DomT4) [inline gnuplot4.rb]: And the spacing between the options and the explanations is a little wide. Should be a single space.

File: gnuplot4.rb [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  option 'qt',     'Build the Qt4 terminal'
+  option 'cairo',  'Build the Cairo based terminals'
+  option 'nolua',  'Build without the lua/TikZ terminal'
+  option 'nogd',   'Build without gd support'
+  option 'tests',  'Verify the build with make check (1 min)'
+  option 'without-emacs', 'Do not build Emacs lisp files'
+  option 'latex',  'Build with LaTeX support'
+  option 'with-aquaterm', 'Build with AquaTerm support'
+
+  deprecated_option "with-x" => "with-x11"
```
Comment (DomT4) [inline gnuplot4.rb]: Kill this line. It's essentially a new formula here, so we haven't deprecated anything yet :)

File: gnuplot4.rb [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+  #  sha1 "c6a2e3f30495c1bd790ea5091f40b7644d695112" => :yosemite
+  #  sha1 "03d507d87eedd8c4bf3e460931081a10403f379d" => :mavericks
+  #  sha1 "24618fd48a6d5fa2f69843da8ac2aaa8d631ff48" => :mountain_lion
+  #end
+
+  option 'pdf',    'Build the PDF terminal using pdflib-lite'
+  option 'wx',     'Build the wxWidgets terminal using pango'
+  option 'qt',     'Build the Qt4 terminal'
+  option 'cairo',  'Build the Cairo based terminals'
+  option 'nolua',  'Build without the lua/TikZ terminal'
```
Comment (DomT4) [inline gnuplot4.rb]: Same thing here with the ` with ` argument, just ` without-lua `.

File: gnuplot4.rb [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+  option 'nogd',   'Build without gd support'
+  option 'tests',  'Verify the build with make check (1 min)'
+  option 'without-emacs', 'Do not build Emacs lisp files'
+  option 'latex',  'Build with LaTeX support'
+  option 'with-aquaterm', 'Build with AquaTerm support'
+
+  deprecated_option "with-x" => "with-x11"
+
+  depends_on 'pkg-config' => :build
+  depends_on LuaRequirement unless build.include? 'nolua'
```
Comment (DomT4) [inline gnuplot4.rb]: And after the changes above, ` if build.with? "lua" ` rather than ` unless build.include? 'nolua' `.

File: gnuplot4.rb [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+  depends_on 'readline'
+  depends_on "libpng"
+  depends_on "jpeg"
+  depends_on "libtiff"
+  depends_on "fontconfig"
+  depends_on 'pango'       if build.include? 'cairo' or build.include? 'wx'
```
Comment (DomT4) [inline gnuplot4.rb]: ` if build.with? ` throughout the block.

File: gnuplot4.rb [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+    gd = Formula["gd"].opt_prefix
+
+    args = %W[
+      --disable-dependency-tracking
+      --disable-silent-rules
+      --prefix=#{prefix}
+      --with-readline=#{readline}
+    ]
+
+    args << "--with-pdf=#{pdflib}" if build.include? 'pdf'
```
Comment (DomT4) [inline gnuplot4.rb]: Same thing throughout here. ` if build.with? ` and so on.

File: gnuplot4.rb [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+      args << '--with-tutorial'
+    else
+      args << '--without-latex'
+      args << '--without-tutorial'
+    end
+
+    system "./configure", *args
+    ENV.j1 # or else emacs tries to edit the same file with two threads
+    system 'make'
+    system 'make check' if build.include? 'tests' # Awesome testsuite
```
Comment (DomT4) [inline gnuplot4.rb]: ` system "make", "check" ` and similar.

File: gnuplot4.rb [hunk 11]
```diff
@@ -0,0 +1,122 @@
+class LuaRequirement < Requirement
+  fatal true
+  default_formula 'lua'
+  satisfy { which 'lua' }
+end
+
+class Gnuplot < Formula
```
Comment (DomT4) [inline gnuplot4.rb]: The name needs changing here to match the formula name, ` gnuplot4` in this case.

**Comment by DomT4** [comment]
Welcome to Homebrew, and thanks for contributing so soon! Just style nits here, nothing awfully wrong. I've pointed out a lot of the changes, and running ` brew audit --strict /path/to/formula ` locally will highlight anything I missed. After that, It should be fine to go in.

**Comment by ppannuto** [comment]
Thanks for the quick and thorough review. The new patch is audit-clean. Some of the changes I pulled in from the newest gnuplot in Homebrew, but I think I did everything correctly.

File: gnuplot4.rb [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+      args << "--with-tutorial"
+    else
+      args << "--without-latex"
+      args << "--without-tutorial"
+    end
+
+    system "./configure", *args
+    ENV.j1 # or else emacs tries to edit the same file with two threads
+    system "make"
+    system "make", "check" if build.include? "tests" # Awesome testsuite
```
Comment (DomT4) [inline gnuplot4.rb]: Missed this one :) - ` if build.with? "tests" `. Kill the comment about the testsuite as well please, Sadly the vast majority of people won't see it.

File: gnuplot4.rb [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+  option "with-x11", "Build with X11 support"
+
+  depends_on "pkg-config" => :build
+  depends_on LuaRequirement if build.with? "lua"
+  depends_on "readline"
+  depends_on "libpng"
+  depends_on "jpeg"
+  depends_on "libtiff"
+  depends_on "fontconfig"
+  depends_on "pango"       if (build.with? "cairo") || (build.with? "wxmac")
```
Comment (DomT4) [inline gnuplot4.rb]: Can you nudge the spacing back on this a bit? So it looks similar to the ` LuaRequirement if build.with? "lua" ` spacing? Thanks!

File: gnuplot4.rb [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+  depends_on :x11 => :optional
+  depends_on "pdflib-lite" => :optional
+  depends_on "gd" => :recommended
+  depends_on "wxmac" => :optional
+  depends_on "qt" => :optional
+  depends_on :tex          if build.with? "latex"
```
Comment (DomT4) [inline gnuplot4.rb]: And spacing here too.

**Comment by DomT4** [comment]
Few more tiny little style nits, but this looks great, thanks for all the changes!

**Comment by ppannuto** [comment]
All fixed. The tests comment wasn't mine, copied from the old gnuplot file :)

**Comment by DomT4** [comment]
Awesome, Thanks! The test-bot is a bit backed up at the moment, but it should get to this later today, and once it gives it the green tick I'll merge this.

Looks like the comment on the testsuite dates back years; Surprised it hasn't been removed in Homebrew/homebrew yet to be honest. Thanks for killing it off here :smiley_cat:

File: gnuplot4.rb [hunk 15]
```diff
@@ -0,0 +1,110 @@
...
+  satisfy { which "lua" }
+end
+
+class Gnuplot4 < Formula
+  homepage "http://www.gnuplot.info"
+  url "https://downloads.sourceforge.net/project/gnuplot/gnuplot/4.6.6/gnuplot-4.6.6.tar.gz"
+  sha256 "1f19596fd09045f22225afbfec11fa91b9ad1d95b9f48406362f517d4f130274"
+
+  option "with-pdf", "Build the PDF terminal using pdflib-lite"
+  option "with-wx", "Build the wxWidgets terminal using pango"
```
Comment (DomT4) [inline gnuplot4.rb]: Gah. Feel free to whisper rude things about me for this, but can you change this line to ` with-wxmac ` so it links up properly with the ` depends_on "wxmac" => :optional ` below? Sorry :crying_cat_face:

**Comment by ppannuto** [comment]
No big deal. Fixed.

**Comment by DomT4** [comment]
Merged! Thanks for your contribution to Homebrew, and taking the time to work through the old style issues and iron everything out - Appreciate that.

**Pull request closed without merging**